WooCommerce/Classes/ViewRelated/BottomSheet/BottomSheetPresenter.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+
+/// Based on `UISheetPresentationController`'s properties for bottom sheet customizations.
+protocol BottomSheetConfigurable {
+    var prefersEdgeAttachedInCompactHeight: Bool { get set }
+    var largestUndimmedDetentIdentifier: UISheetPresentationController.Detent.Identifier? { get set }
+    var prefersGrabberVisible: Bool { get set }
+    var detents: [UISheetPresentationController.Detent] { get set }
+}
+
+extension UISheetPresentationController: BottomSheetConfigurable {}
+
+/// Handles presentation and dismissal of a bottom sheet natively.
+final class BottomSheetPresenter: NSObject {
+    private var viewController: UIViewController?
+    private var onDismiss: (() -> Void)?
+
+    private let configure: ((BottomSheetConfigurable) -> Void)
+
+    /// - Parameter configure: Customizations of the bottom sheet with a default implementation.
+    init(configure: @escaping ((BottomSheetConfigurable) -> Void) = { bottomSheet in
+        var sheet = bottomSheet
+        sheet.prefersEdgeAttachedInCompactHeight = true
+        sheet.largestUndimmedDetentIdentifier = .large
+        sheet.prefersGrabberVisible = false
+        sheet.detents = [.medium()]
+    }) {
```

Comment:
Nit: Should we move this to a static closure instead putting it inline to make it more readable?

```suggestion
    private static let defaultConfig: ((BottomSheetConfigurable) -> Void) = { bottomSheet in
        var sheet = bottomSheet
        sheet.prefersEdgeAttachedInCompactHeight = true
        sheet.largestUndimmedDetentIdentifier = .large
        sheet.prefersGrabberVisible = false
        sheet.detents = [.medium()]
    }
    init(configure: @escaping ((BottomSheetConfigurable) -> Void) = Self.defaultConfig) {
```

Reply:
The use of `Self` isn't allowed in the default parameter value, I moved it to a private enum `Defaults` in https://github.com/woocommerce/woocommerce-ios/pull/9552/commits/984110a24a1f880610653401700aa7007a9c0f73